internal/app/models.go

```diff
@@ -0,0 +1,41 @@
+package app
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Vault represents a vault.
+type Vault string
```

Comment:
Have we now killed the "namespace.relation" semantics?

Reply:
not yet, we can't kill it yet because the provider kind of depends on this semantics to create the namespace entry in the database. We can only kill it after remodeling the database tables (I'm planning to write a doc on how to do that remodeling, shouldn't be hard)

it's being enforced [here](https://github.com/tablelandnetwork/basin-cli/pull/28/files#diff-73777451ff3bdcfcd6ee6970408801e8731a6368a183dfca6d1e1b7df2195fc7L110) in the CLI  and [here](https://github.com/tablelandnetwork/basin-provider/blob/main/lib/worker/src/domain/vault.rs#L12-L13) in the provider